Library/Formula/boost.rb

```diff
@@ -36,6 +36,10 @@ class Boost < Formula
     cause "Dropped arguments to functions when linking with boost"
   end
 
+  # Patch for odeint. Can (hopefully) be removed with the next release of Boost (1.58).
+  # See https://github.com/headmyshoulder/odeint-v2/issues/138
+  patch :DATA
```

Comment:
Could you try and open a pull request? This should help and I'm happy to walk you through anything else: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged

Thanks!